src/buffer/cell.rs

```diff
@@ -34,6 +34,39 @@ pub struct Cell {
 }
 
 impl Cell {
+    /// An empty `Cell`
+    pub const EMPTY: Self = Self::new_inline(" ");
+
+    /// Creates a new `Cell` with the given symbol.
+    pub fn new(symbol: &str) -> Self {
+        Self {
+            symbol: CompactString::new(symbol),
+            fg: Color::Reset,
+            bg: Color::Reset,
+            #[cfg(feature = "underline-color")]
+            underline_color: Color::Reset,
+            modifier: Modifier::empty(),
+            skip: false,
+        }
+    }
+
+    /// Creates a new `Cell` with the given symbol.
+    ///
+    /// This works at compile time and puts the symbol onto the stack. Fails to build when the
+    /// symbol doesnt fit onto the stack and requires to be placed on the heap. Use [`Self::new`] in
+    /// that case. See [`CompactString::new_inline`] for more details on this.
+    pub const fn new_inline(symbol: &str) -> Self {
```

Comment:
Would it be the worst thing to make this the default (i.e. `new()`) and just not support creating new Cells with long symbols that can't be stored on the stack without using `Cell::default().set_symbol()`? I suspect that there is rarely if ever a case where the cell can't be stored on the stack (long ANSI strings spring to mind as a counterpoint).

Reply:
Especially as its only the creation. Normally set_symbol is used on the existing ones on the existing buffer.
So yeah, can be the standard new method.